src/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -253,5 +278,67 @@ static CompletionFilterReason GetFilterReason(CompletionTrigger trigger)
                 };
             }
         }
+
+        private static MatchResult GetBestCompletionItemSelectionFromFilteredResults(IReadOnlyList<MatchResult> filteredMatchResults)
```

Comment:
is this copied from ItemManager? We uses this to actually determine the selected item in VS, but here this is at best a heuristic which might not match VSCode selection.

Reply:
Yes this is copied.  Understood that it is a heuristic which may not match, but this definitely works better than before

Reply:
reverted